addons/mail/models/mail_alias_domain.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, exceptions, fields, models, _
+
+
+class AliasDomain(models.Model):
+    """ Model alias domains, now company-specific. Alias domains are email
+    domains used to receive emails through catchall and bounce aliases, as
+    well as using mail.alias records to redirect email replies.
+
+    This replaces ``mail.alias.domain`` configuration parameter use until v16.
+    """
+    _name = 'mail.alias.domain'
+    _description = "Email Domain"
+    _order = 'sequence ASC, id DESC'
+    _rec_name = 'name'
+
+    name = fields.Char('Name', required=True)
+    company_ids = fields.One2many(
```

Comment:
Strange, commit seems to mention a many2many relationship but it's not really the case?